libs/traits/src/liquidity_pools.rs

```diff
@@ -36,16 +33,38 @@ pub trait LPEncoding: Sized {
 	fn empty() -> Self;
 }
 
-/// The trait required for sending outbound messages.
-pub trait Router {
-	/// The sender type of the outbound message.
-	type Sender;
+pub trait RouterSupport<Domain>: Sized {
```

Comment:
One nit though - maybe we can adjust this a bit so we don't have to implement it for the `RouterId`, which also has the potential adding more to the confusion.

How about something like:

```
pub trait RouterProvider<Domain>: Sized {
	type RouterId;
	
	/// Returns a list of router IDs supported for the given domain.
	fn for_domain(domain: Domain) -> Vec<Self::RouterId>;
}
```

Then we can have a simple enum as a router ID, and a different type for getting routers for a domain, all in runtime common.

WDYT @lemunozm ?

Reply:
I like it! I was not comfortable with the correct way. You can change this in your PR, or we can do it after merging everything, as a clean PR. As you want.

Reply:
Maybe `for_domain()` can also be renamed to be more specific: `routers_for_domain()`

Reply:
I'll add these in my PR to have a clean state there. Thank you!